compiler/src/dotty/tools/dotc/core/Types.scala

```diff
@@ -3580,13 +3650,35 @@ object Types {
   /** The parameter of a type lambda */
   case class LambdaParam(tl: TypeLambda, n: Int) extends ParamInfo {
     type ThisName = TypeName
+
     def isTypeParam(implicit ctx: Context): Boolean = tl.paramNames.head.isTypeName
     def paramName(implicit ctx: Context): tl.ThisName = tl.paramNames(n)
     def paramInfo(implicit ctx: Context): tl.PInfo = tl.paramInfos(n)
     def paramInfoAsSeenFrom(pre: Type)(implicit ctx: Context): tl.PInfo = paramInfo
     def paramInfoOrCompleter(implicit ctx: Context): Type = paramInfo
-    def paramVariance(implicit ctx: Context): Int = tl.paramNames(n).variance
     def paramRef(implicit ctx: Context): Type = tl.paramRefs(n)
+
+    private var myVariance: FlagSet = UndefinedFlags
+    def storedVariance_= (v: Variance): Unit =
+      myVariance = v
+    def storedVariance: Variance =
+      myVariance
+
+    def givenVariance_=(v: Variance): Unit =
+      assert(myVariance == UndefinedFlags)
+      myVariance = v
+    def givenVariance: Variance =
```

Comment:
This method name is misleading as is, since it can be called after `storedVariance` has been set and will return the stored variance. Since this really should only be called on lambdas with given variance, we could prevent this by doing:
```suggestion
    def givenVariance: Variance =
      assert(tl.isVariantLambda)
```
And documenting the precondition appropriately.